packages/calypso-e2e/src/lib/pages/stats-page.ts

```diff
@@ -1,13 +1,31 @@
-import { Page } from 'playwright';
+import { Locator, Page } from 'playwright';
+import { envVariables } from '../..';
+import { getCalypsoURL } from '../../data-helper';
 import { clickNavTab } from '../../element-helper';
 
-export type StatsTabs = 'Traffic' | 'Insights' | 'Store';
+export type StatsTabs = 'Traffic' | 'Insights' | 'Subscribers' | 'Store';
+type TrafficActivityType = 'Views' | 'Visitors' | 'Likes' | 'Comments';
+type StoreActivityType = 'Gross Sales' | 'Net sales' | 'Orders' | 'Avg. Order Value';
+// Discriminated Union type.
+type ActivityTypes =
+	| { tab: 'Traffic'; type: TrafficActivityType }
+	| { tab: 'Store'; type: StoreActivityType };
```

Comment:
This is the first time I'm using a discriminated union type as parameter. 

I settled on this approach after searching high and low for ways to narrow a union type. Common suggestions on the web starts involving things like helper functions or generics, which is (frankly) stupid. 

We should be able to do something like this and it _should just work_:

```typesccript
if (typeof activityType === TrafficActivityType)
```

But it doesn't work, so here I am with the discriminated union.

Reply:
FWIW -- I think this is a great use of a discriminated union type and reads really well here!